### ISSUE TITLE
SOW 2 - Task 2.6:  Modify Summarize Releases

### ISSUE THREAD

**Issue by jasmyace** [tmcd82070/CAMP_RST#28]
Modify the Summarize releases report and any underlying trap efficiency data in the production reports so each trap efficiency test's ReleaseID is reflected as a unique item.

**Comment by jasmyace** [comment]
(A recap of an email conversation between Jason and Doug, June 2015)



Jason:
 
I went back and looked at the Summarize by Release report.
 
The Good News:  As a follow-up to the email you sent last Friday, I wanted to validate the development of a horizontal trap efficiency line in the efficiency PNG.  So I went back and manually calculated the mean trap efficiency for the South Channel 8 trap on the American River in 2013 using a spreadsheet the biologist maintains.  If I use your approach to calculate trap efficiency and I use the biologist’s data, I get the same trap efficiency, i.e. 1.04%, which is what is also reflected in the PNG file (see below).  So I used a different method to try and calculate a mean trap efficiency when the GAM uses that kind of model and I came up with the same result.
 
![image003 1](https://cloud.githubusercontent.com/assets/9057972/12272961/5bfca370-b920-11e5-9fc2-4688ed83406c.jpg)

The  South Channel 8 trap has the light green triangles.   Note that the line intersects the Y axis at about .01 (1%).

![image004](https://cloud.githubusercontent.com/assets/9057972/12272965/63c5d572-b920-11e5-8a75-75141035f150.jpg)

The Bad News:  I just spotted something in the Summarize by Release report that isn’t correct.  See the screen shot below.  For row number 11, you will note that there is a TrapPosition = NorthChannel 8.1  with nRecaps = 36 for releaseID1.  There were not 36 recaps in Trap NorthChannel 8.1, and that 36 pertains to the total # of recaps for all traps running that day (there were 2 traps running that day). Hence the Releasecomment from the biologist that says “Total recaps =36”.  I checked the biologist’s trap efficiency spreadsheet and they have 29 recaps in Trap 8.1 and 7 recaps in Trap 8 South.  You’ll also note there are no data for the 4 other traps that were operated on the American River in the CSV.  Please revise the R code for that report so: (1) the trap efficiency data for all the traps that operated in the specified date range are provided (there were 5 American River RSTs running in 2013, not just NorthChannel 8.1)  and (2) the nRecaps number is specific to each trap. 
 
 ![image005](https://cloud.githubusercontent.com/assets/9057972/12272974/706d891e-b920-11e5-9681-bc073f7bf2e0.jpg)

Much more concerning are problems relating to traps 5 North and 8.2.  see the attached PDF.  The PNG trap efficiency plot shows 8 blue dots for Trap 8.2, and 3 green diamonds for Trap 5 North.  There should only be 7 and 2 data points for those traps, respectively. Those extra dots are highlighted in yellow, and they each have a trap efficiency proportion = 0.00.  For some reason, R “thinks” there is 1 extra data point for each trap, and those extra data points are skewing the trap efficiency efficiencies for each trap, especially the 8.2 trap.  See the screen shot below and compare it to the PDF.  If I use your approach to calculate the trap efficiency for trap 8.2, I should get 2.24%, but your new trap efficiency CSV is yielding 1.12% due to the extra data point for trap 8.2.  Connie called me today to say she had found a problem with the Marks Existing table in the CAMP.mdb file.  I do not know if the problem she detected today is related to the problem that is showing up in the trap efficiency R code/outputs.  The problem described in this paragraph is probably related to data in the CAMP.mdb and not the R code.  Having the updated R code as described in the paragraph above will help narrow down where the problem is and that Connie and the biologists will have to deal with.  The new trap efficiency CSV is helpful because it can be used to detect a previously unknown problem that has likely existed in the American River data for a long time.
 
![image012](https://cloud.githubusercontent.com/assets/9057972/12272987/83d652ce-b920-11e5-9420-73ef79b6c794.jpg)

[DOC048.PDF](https://github.com/tmcd82070/CAMP_RST/files/87795/DOC048.PDF)

**Comment by jasmyace** [comment]
Doug indicated in his email above that "The problem described in this paragraph is probably related to data in the CAMP.mdb and not the R code."  This appears to be true.  When I run ALL passage for this river and time frame, the efficiency looks to be as Doug indicates it should be.  Note how the efficiency for trap 8.2 (the blue) is a horizontal line at approximately 2.42%, where the calculations on the ad hoc biologist-focused embedded spreadsheet above indicates it should be.  (Note that Doug writes in the text that this number should be 2.24%, but I think he transposed the numbers.)

![task25_lar_2016-01-12_12-05-55fall_eff](https://cloud.githubusercontent.com/assets/9057972/12274133/a137cb8a-b926-11e5-9d8e-caca4f6109fe.png)

I then ran the Summarize Releases report to see what it reports.  It reports (with no effort on my part) exactly what Doug seems to want:  it breaks out the trapping effort on February 3rd between the two traps that operated that day, with 29 nRecaps for the one, and 7 for the other.  Also note that all five traps are in the list at least once.  This seems to be what Doug describes what he wishes to see.  

![image](https://cloud.githubusercontent.com/assets/9057972/12274405/039a1368-b928-11e5-9d78-381dbad2268e.png)


In the original email exchange, Doug writes that Connie was going to do some stuff to the American database.  It seems that whatever Connie did ameliorated the issue that was preventing the Summarize Releases report from displaying the appropriate data.  

Let me know if something more needs to be done here, or if there is another river or time frame for which some testing should occur.  Otherwise, it seems this Issue has been made obsolete.

**Comment by dougthreloff** [comment]
Jason:



Yes, I think issue 2.6 has been resolved. = 1 more item done.



Thanks.



Doug



*From:* Jason [mailto:notifications@github.com]
*Sent:* Tuesday, January 12, 2016 11:29 AM
*To:* tmcd82070/CAMP_RST
*Subject:* Re: [CAMP_RST] SOW 2 - Task 2.6: Modify Summarize Releases (#28)



Doug indicated in his email above that "The problem described in this
paragraph is probably related to data in the CAMP.mdb and not the R code."
This appears to be true. When I run ALL passage for this river and time
frame, the efficiency looks to be as Doug indicates it should be. Note how
the efficiency for trap 8.2 (the blue) is a horizontal line at
approximately 2.42%, where the calculations on the ad hoc biologist-focused
embedded spreadsheet above indicates it should be. (Note that Doug writes
in the text that this number should be 2.24%, but I think he transposed the
numbers.)

[image: Image removed by sender. task25_lar_2016-01-12_12-05-55fall_eff]
<https://cloud.githubusercontent.com/assets/9057972/12274133/a137cb8a-b926-11e5-9d8e-caca4f6109fe.png>

I then ran the Summarize Releases report to see what it reports. It reports
(with no effort on my part) exactly what Doug seems to want: it breaks out
the trapping effort on February 3rd between the two traps that operated
that day, with 29 nRecaps for the one, and 7 for the other. Also note that
all five traps are in the list at least once. This seems to be what Doug
describes what he wishes to see.

[image: Image removed by sender. image]
<https://cloud.githubusercontent.com/assets/9057972/12274405/039a1368-b928-11e5-9d78-381dbad2268e.png>

In the original email exchange, Doug writes that Connie was going to do
some stuff to the American database. It seems that whatever Connie did
ameliorated the issue that was preventing the Summarize Releases report
from displaying the appropriate data.

Let me know if something more needs to be done here, or if there is another
river or time frame for which some testing should occur. Otherwise, it
seems this Issue has been made obsolete.

—
Reply to this email directly or view it on GitHub
<https://github.com/tmcd82070/CAMP_RST/issues/28#issuecomment-171025692>.[image:
Image removed by sender.]
